src/libtexture/imagecache.cpp

```diff
@@ -2763,7 +2761,7 @@ ImageCacheImpl::get_image_info(ImageCacheFile* file,
     if (file->broken()) {
         if (file->errors_should_issue()) {
             // eat any earlier messages
-            imcache_error_messages.erase(this);
+            imcache_error_messages.erase(imagecache_id_atomic);
```

Comment:
I think you meant to use `imagecache_id` here?

Maybe rename `imagecache_id_atomic` with `imagecache_next_id` to avoid confusion?

Reply:
Quite right. Fix pushed.